Automattic/jetpack
# Protect: Add WAF StatCards

## Description

## Proposed changes:

* Makes the `StatCard` js-packages component easily accessible
* Adds `Waf_Blocklog_Manager` class and migrates related methods over 
* Adds a new set of methods for tracking daily and monthly blocked requests stats via new `jetpack_waf_blocklog_daily_summary`,  and `jetpack_waf_all_time_stats` options
* Adds the `StatCard` components to the `FirewallHeader` and hooks up the new stats

### Other information:


## Jetpack product discussion


- https://github.com/Automattic/jetpack-scan-team/issues/381

## Does this pull request change what data or activity we track or use?


- No

## Testing instructions:






* Check out this branch and start up your Jurassic Tube with Protect installed and active with a scan-level plan
* Enable the WAF, ensure share data is toggled, configure your settings as desired
* Register several blocked requests and verify that the stats in the UI are updated accordingly on refresh
* Modify the `$date` value in the `update_daily_summary()` method of the `Waf_Blocklog_Manager` class to simulate requests from a different day within the same month and from outside of the current month
* Verify that the `jetpack_waf_blocklog_daily_summary`, and `jetpack_waf_all_time_stats` option data and stats in the UI reflect the entries and that any data logged outside of the current month is pruned
* Verify that stats are 0 and disabled when a scan upgrade is removed (paid feature flag is set) and/or share data is disabled
* Ensure `jetpack-waf/waf-blocklog` entries or `jetpack_waf_blocklog_daily_summary`, and `jetpack_waf_all_time_stats` option updates are made when share data is disabled (note that `wp_jetpack_waf_blocklog` table entries will continue as well)
* Modify the option data to verify display when values are significant
* Switch between desktop and mobile view to ensure the display is responsive
* Ensure no regressions are introduced relating to WAF and blocklog operations
* Run `cd projects/js-packages/storybook && npm run storybook:dev` and verify that all Protect `FirewallHeader` stories render
* Run `jetpack test packages/waf php` to ensure that all integration tests pass

## Screenshots:
Desktop:
<img width="1152" alt="Screen Shot 2024-02-22 at 15 07 29" src="https://github.com/Automattic/jetpack/assets/43220201/04658682-3958-47c4-9e31-6ac2f4a98cdb">

Mobile:
<img width="342" alt="Screen Shot 2024-02-22 at 15 08 15" src="https://github.com/Automattic/jetpack/assets/43220201/8ce12961-fec0-486b-a75d-0a1f0048ead4">
